R/pkg/R/generics.R

```diff
@@ -748,7 +748,7 @@ setGeneric("add_months", function(y, x) { standardGeneric("add_months") })
 
 #' @rdname column_aggregate_functions
 #' @name NULL
-setGeneric("approxCountDistinct", function(x, ...) { standardGeneric("approxCountDistinct") })
+setGeneric("approx_count_distinct", function(x, ...) { standardGeneric("approx_count_distinct") })
```

Comment:
@felixcheung might want to check if I'm handling these R changes correctly

Reply:
my concern is that these are breaking changes in a version without having them deprecated first...
could we leave the old one to redirect and add .Deprecate?

Reply:
I think my comment didn't get connected to this one -- @felixcheung what do you think about the argument that this almost surely was meant to be deprecated along with counterparts in Scala/Python? leaving them in would make this inconsistent. As the degrees, radians, and approxCountDistinct are reasonably niche and have a direct replacement that's compatible with older versions, I feel like this is OK for 3.0?

Reply:
I think it's super light weight to have a `approxCountDistinct` that calls `approx_count_distinct` with deprecation?
I thought was that R API was not always sync or complete compare to python, and a breaking API change - ie. the job will fail - seems a bit drastic even in a major release.

Reply:
It is, but then again that's exactly what was deprecated and removed in Python and Scala. Major versions can have breaking changes. Yes R isn't always in sync but that's a bug not a feature? Let me surface this to dev@ as I think it's going to come up a few more times.